### DISCUSSION TITLE
Support for grpc or graphql in Serverpod

### DISCUSSION THREAD

**Discussion by braj065** [serverpod/serverpod#824] [Q&A] [answered]
Hi,

Will grpc or graphql support be available in coming future? Not very nearby future.

**Comment by vlidholt** [comment, marked as answer]
We will most likely add support for the document based features in Postgres before we add support for something like MongoDB. gRPC is a maybe for the future, some of Serverpod's features may be hard to implement through the gRPC protocol. We instead plan to add support for more platforms, such as JS, Kotlin, and Swift.

**Reply by braj065** [reply]
Thanks @vlidholt. For microservices implementation grpc can provide performane benefits. No pressure if that is not a priority. Also will Serverpod implement and provide microservice  features like service discovery etc?

**Reply by vlidholt** [reply]
When you are talking about the performance benefits, are you talking about the efficiency of the protocol vs JSON? We may be adding a binary protocol for this purpose in the future, it may or may not be based upon gRPC. We haven't really done the research for what is best yet. :)

**Reply by braj065** [reply]
Thanks @vlidholt. Will Serverpod support microservices creation, communication, logging(datadog/sentry) and actuator on the microservices cluster? If yes documentation on how to achieve same would be great. Also 1.1 release is dope, Congrats on the work.

**Reply by baudee** [reply]
@vlidholt Looking forward to the document based support! Any idea when it will be available?

**Reply by vlidholt** [reply]
Hopefully sometime later this year! There are a lot of things on our todo-list and we want to do everything really solid. :)
